pkg/scheduler/framework/preemption/preemption.go

```diff
@@ -416,14 +416,17 @@ func nodesWherePreemptionMightHelp(nodes []*framework.NodeInfo, m framework.Node
 	var potentialNodes []*framework.NodeInfo
 	nodeStatuses := make(framework.NodeToStatusMap)
 	for _, node := range nodes {
-		name := node.Node().Name
+		nodeName := node.Node().Name
 		// We rely on the status by each plugin - 'Unschedulable' or 'UnschedulableAndUnresolvable'
-		// to determine whether preemption may help or not on the node.
-		if m[name].Code() == framework.UnschedulableAndUnresolvable {
-			nodeStatuses[node.Node().Name] = framework.NewStatus(framework.UnschedulableAndUnresolvable, "Preemption is not helpful for scheduling")
-			continue
+		// to determine whether preemption may help or not on the node. We first check that the entry
+		// is in the map, as a missing entry implies UnschedulableAndUnresolvable.
+		if status, ok := m[nodeName]; ok {
+			if status.Code() == framework.UnschedulableAndUnresolvable {
+				nodeStatuses[nodeName] = framework.NewStatus(framework.UnschedulableAndUnresolvable, "Preemption is not helpful for scheduling")
+				continue
+			}
+			potentialNodes = append(potentialNodes, node)
```

Comment:
Forgive me if this is a dumb question:
Since we only add the node in framework.NodeToStatusMap to potentialNodes.
Why don't we iterate `framework.NodeToStatusMap` directly instead of iterating `allNodes` and check if the node is in `framework.NodeToStatusMap` ?

Reply:
On the contrary, your comment is absolutely right :)

If we decide to not fill in the map (depending on resolution of https://github.com/kubernetes/kubernetes/pull/125197#discussion_r1622005364), I will implement this

Reply:
So, in this PR, we are still filling the map to minimize changes and facilitate cherry-picking.
However, in version 1.31, we will change the behavior so that we no longer fill the map with UnschedulableAndUnresolvable status, am I understanding correctly??

Reply:
Correct, the `NodeToStatusMap` returned by `nodesWherePreemptionMayHelp`. `UnschedulableAndUnresolvable` status is only used for the error message (#nodes, and #"Preemption is not helpful for scheduling"). I think we can pipe this information in a more efficient way